flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/optimize/ScanReuseTest.java

```diff
@@ -324,4 +326,68 @@ public void testProjectReuseWithPartitionPushDown() {
             util.verifyExecPlan(sqlQuery);
         }
     }
+
+    @Test
+    public void testReuseWithReadMetadataAndWatermarkPushDown1() {
+        assumeTrue(isStreaming);
+        String ddl =
+                "CREATE TABLE MyTable1 (\n"
+                        + "  metadata_0 int METADATA VIRTUAL,\n"
+                        + "  a0 int,\n"
+                        + "  a1 int,\n"
+                        + "  ts STRING,\n "
+                        + "  rowtime as TO_TIMESTAMP(`ts`),\n"
+                        + "  WATERMARK FOR rowtime AS rowtime - INTERVAL '1' SECOND\n"
+                        + ") WITH (\n"
+                        + " 'connector' = 'values',\n"
+                        + " 'bounded' = 'false',\n"
+                        + " 'readable-metadata' = 'metadata_0:int',\n"
+                        + " 'enable-watermark-push-down' = 'true',\n"
+                        + " 'disable-lookup' = 'true'"
+                        + ")";
+        util.tableEnv().executeSql(ddl);
+
+        // join left side value source without projection spec.
+        String sqlQuery =
+                "SELECT T1.a0, T1.a1 FROM"
+                        + " (SELECT MIN(a0) as a0, MIN(a1) as a1, MIN(metadata_0) as metadata_0 "
+                        + " FROM MyTable1 GROUP BY"
+                        + " TUMBLE(rowtime, INTERVAL '1' SECOND)) T1,"
+                        + " (SELECT MIN(a0) as a0 FROM MyTable1 GROUP BY"
+                        + " TUMBLE(rowtime, INTERVAL '1' SECOND)) T2"
```

Comment:
Tumble is deprecated, it would better use the new window tvf

Reply:
> Tumble is deprecated, it would better use the new window tvf

Done!